text_generation/causal_lm/cpp/continuous_batching/library/src/generation_config.cpp

```diff
@@ -7,6 +7,32 @@
 
 #include "generation_config.hpp"
 
+#include "openvino/core/except.hpp"
+
+void GenerationConfig::set_eos_token_id(size_t tokenizer_eos_token_id) {
+    if (eos_token_id < 0) {
+        eos_token_id = tokenizer_eos_token_id;
+    } else {
+        OPENVINO_ASSERT(eos_token_id == tokenizer_eos_token_id,
+            "EOS token ID is different in generation config (", eos_token_id, ") and tokenizer (",
+            tokenizer_eos_token_id, ")");
+    }
+}
+
+void GenerationConfig::validate() const {
+    if (is_multinomial()) {
+        OPENVINO_ASSERT(top_p > 0.0f && top_p <= 1.0f, "top_p must be in the interval (0, 1]");
+        OPENVINO_ASSERT(temperature >= 0.0f, "temperature must be a positive value");
+        OPENVINO_ASSERT(repetition_penalty >= 0.0f, "repetition penalty must be a positive value");
+        OPENVINO_ASSERT(frequence_penalty >= -2.0f && frequence_penalty <= 2.0f, "frequence_penalty penalty must be a [-2; +2]");
+        OPENVINO_ASSERT(presence_penalty >= -2.0f && presence_penalty <= 2.0f, "presence_penalty penalty must be a [-2; +2]");
```

Comment:
are these penalties valid for beam search case?
If yes, let's enable them in a separate PR